src/transactions/OfferExchange.cpp

```diff
@@ -1491,6 +1491,13 @@ convertWithOffers(
     wheatReceived = 0;
 
     bool needMore = (maxWheatReceive > 0 && maxSheepSend > 0);
+    if (needMore && maxOffersToCross == 0 &&
+        ltxOuter.loadHeader().current().ledgerVersion >= 18)
+    {
+        // offerTrail is going to be too long, fast fail
```

Comment:
Can you extend this comment to indicate that this can happen for path payment, but cannot happen for manage offer? I think this is important because if this _could_ happen for manage offer, then it would be incorrect.

Reply:
good idea, done. I also added the same comment in the pool exchange function